### PR TITLE
Python 3: Add RPM package support

### PR DESCRIPTION
**Pull request by clebergnu**
This PR brings support for Python 3 versions of the Avocado RPM packages.

### CODE REVIEW

**Comment by clebergnu** [comment]
[Python 3: Avocado RPM packaging](https://trello.com/c/XdWoIQ1E/1268-python-3-avocado-rpm-packaging)

File: python-avocado.spec [hunk 1]
```diff
@@ -350,8 +344,6 @@ Summary: Avocado Runner for Execution on Docker Containers
 Requires: %{name} == %{version}
 Requires: %{name}-plugins-runner-remote == %{version}
 Requires: python-aexpect
-Obsoletes: %{srcname}-plugins-runner-docker < 47.0-1
-Provides: %{srcname}-plugins-runner-docker = %{version}-%{release}
```
Comment (apahim) [inline python-avocado.spec]: oh, ok

File: python-avocado.spec [hunk 2]
```diff
@@ -311,10 +311,6 @@ arbitrary filesystem location.
 Summary: Avocado Runner for Remote Execution
 Requires: %{name} == %{version}
 Requires: fabric
-%if 0%{?fedora} == 24
-Requires: python-crypto
-BuildRequires: python-crypto
-%endif
```
Comment (apahim) [inline python-avocado.spec]: ok, F24 is long gone.

File: python-avocado.spec [hunk 3]
```diff
@@ -476,6 +477,7 @@ Again Shell code (and possibly other similar shells).
 * Mon Mar 19 2018 Cleber Rosa <cleber@redhat.com> - 59.0-2
 - Removed backward compatibility with name avocado on plugins
 - Removed extra dependencies on Fedora 24 for runner-remote
+- Added python-avocado requirement for golang plugin
```
Comment (apahim) [inline python-avocado.spec]: ok

File: python-avocado.spec [hunk 4]
```diff
@@ -294,6 +280,29 @@ AVOCADO_CHECK_LEVEL=0 selftests/run
...
+%config(noreplace)%{_sysconfdir}/avocado/avocado.conf
+%config(noreplace)%{_sysconfdir}/avocado/conf.d/README
+%config(noreplace)%{_sysconfdir}/avocado/conf.d/gdb.conf
+%config(noreplace)%{_sysconfdir}/avocado/conf.d/jobscripts.conf
+%config(noreplace)%{_sysconfdir}/avocado/sysinfo/commands
+%config(noreplace)%{_sysconfdir}/avocado/sysinfo/files
+%config(noreplace)%{_sysconfdir}/avocado/sysinfo/profilers
+%config(noreplace)%{_sysconfdir}/avocado/scripts/job/pre.d/README
+%config(noreplace)%{_sysconfdir}/avocado/scripts/job/post.d/README
+
```
Comment (apahim) [inline python-avocado.spec]: ok, I see where you're heading.

File: python-avocado.spec [hunk 5]
```diff
@@ -203,8 +203,8 @@ popd
 %{__cp} -r examples/yaml_to_mux %{buildroot}%{_datadir}/avocado
 %{__cp} -r examples/yaml_to_mux_loader %{buildroot}%{_datadir}/avocado
 %{__cp} -r examples/varianter_pict %{buildroot}%{_datadir}/avocado
-%{__mkdir} -p %{buildroot}%{_libexecdir}/
-ln -s %{python_sitelib}/avocado/libexec/ %{buildroot}%{_libexecdir}/avocado
+%{__mkdir} -p %{buildroot}%{_libexecdir}/avocado
+%{__mv} %{buildroot}%{python_sitelib}/avocado/libexec/* %{buildroot}%{_libexecdir}/avocado
```
Comment (apahim) [inline python-avocado.spec]: ok

File: python-avocado.spec [hunk 6]
```diff
@@ -382,75 +397,75 @@ Requires: PyYAML
...
+%description -n python2-%{srcname}-plugins-glib
 This optional plugin is intended to list and run tests written in the
 GLib Test Framework.
 
-%files plugins-glib
-%{python_sitelib}/avocado_glib*
-%{python_sitelib}/avocado_framework_plugin_glib*
+%files -n python2-%{srcname}-plugins-glib
+%{python2_sitelib}/avocado_glib*
+%{python2_sitelib}/avocado_framework_plugin_glib*
```
Comment (apahim) [inline python-avocado.spec]: ok, testing all this.

**Comment by clebergnu** [comment]
@apahim it's the temporary directories issue... and it gets uglier to have a successful run the more commits we have.  So, I'm merging this based on your review and approval.

**Pull request merged**